### PR TITLE
Track device automatically

### LINKED ISSUE #122: Track device automatically.
Track `self.device = device` automatically as a property of the API class

### PR DESCRIPTION
**Pull request by aniketmaurya**
fixes #122

### CODE REVIEW

File: src/litserve/server.py [hunk 1]
```diff
@@ -254,6 +254,7 @@ def inference_worker(
     stream,
 ):
     lit_api.setup(device)
+    lit_api.device = device
```
Comment (aniketmaurya) [inline src/litserve/server.py:257]: should we have a single method to do both of these? maybe in future we can have `litapi.trigger_setup` which is mean to be used by LitServer to trigger methods.

Comment (lantiga) [reply src/litserve/server.py:257]: Sorry I didn't get this, can you clarify?

**Review by lantiga — APPROVED** [review]
Looks good. In the future we could remove `device` as an argument of `setup`.
Not that it hurts for now, so let's keep it, but we'll have to consider it as we introduce non-PyTorch examples, since not all frameworks have the concept of device.
And maybe as we get into serving the model from multiple devices using tensor or sequential parallelism.

File: src/litserve/server.py (hunk 1 above)
Comment (aniketmaurya) [reply src/litserve/server.py:257]: my bad, I thought that I selected two lines while adding the comment. 
So right now, we call `litapi.setup` and then add set `litapi.device`, which kinda creates two source of truth. 

```
lit_api.setup(device)
lit_api.device = device
```

Comment (aniketmaurya) [reply src/litserve/server.py:257]: saw your other comment, I think it won't matter when we remove device from setup.

**Pull request merged**